api/src/transactions.rs

```diff
@@ -1376,7 +1373,16 @@ impl TransactionsApi {
         let version = ledger_info.version();
 
         // Ensure that all known statuses return their values in the output (even if they aren't supposed to)
-        let exe_status = ExecutionStatus::convert_vm_status_for_simulation(vm_status.clone());
+        let default_on_features = Features::default();
+        let exe_status = if default_on_features.is_enabled(FeatureFlag::REMOVE_DETAILED_ERROR_FROM_HASH){
+            ExecutionStatus::convert_vm_status_for_simulation(vm_status.clone())
+        } else {
+            match output.status().clone() {
+                TransactionStatus::Keep(exec_status) => exec_status,
+                TransactionStatus::Discard(status) => ExecutionStatus::MiscellaneousError(Some(status)),
+                _ => ExecutionStatus::MiscellaneousError(None),
+            }
```

Comment:
So if the transaction status is discarded, we return misc error.

When is a situation that it's discarded, so if I was developing / I was a dapp user to ensure I don't get a misc error?

Reply:
Getting misc error for the discarded transactions is the behavior before my change `https://github.com/aptos-labs/aptos-core/commit/05f4cbfe1d62cf05b679d49cdd5b355db0d14ca0#diff-48b9818e00a894aa17300f3da5839340a5e93195e7c95c6395e4347a7bf1a233`.  Failed transaction can be kept with abort info.